diff --git a/src/motiondetect.c b/src/motiondetect.c
index af96772..28d0b77 100644
--- a/src/motiondetect.c
+++ b/src/motiondetect.c
@@ -725,12 +725,14 @@ LocalMotions calcTransFields(VSMotionDetect* md,
 #endif
 
   VSVector goodflds = selectfields(md, fields, contrastfunc);
+  const int numfields = vs_vector_size(&goodflds);
+  int index;
   // use all "good" fields and calculate optimal match to previous frame
 #ifdef USE_OMP
   omp_set_num_threads(md->conf.numThreads);
 #pragma omp parallel for shared(goodflds, md, localmotions)
 #endif
-  for(int index=0; index < vs_vector_size(&goodflds); index++){
+  for(index=0; index < numfields; index++){
     int i = ((contrast_idx*)vs_vector_get(&goodflds,index))->index;
     LocalMotion m;
     m = fieldfunc(md, fields, &fields->fields[i], i); // e.g. calcFieldTransPlanar
diff --git a/src/serialize.c b/src/serialize.c
index 2b0a1b2..7e27365 100644
--- a/src/serialize.c
+++ b/src/serialize.c
@@ -58,7 +58,7 @@ static double byteSwapDouble(double v)
 }
 #elif defined(__BYTE_ORDER) && __BYTE_ORDER == __LITTLE_ENDIAN || \
     defined(__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__ || \
-    defined(__LITTLE_ENDIAN__) || \
+    defined(__LITTLE_ENDIAN__) || defined(_M_IX86) || defined(_M_X64) || \
     defined(__ARMEL__) || \
     defined(__THUMBEL__) || \
     defined(__AARCH64EL__) || \
@@ -386,7 +386,7 @@ int vsReadFileVersionBinary(FILE* f){
   unsigned char version;
   VSMotionDetectConfig conf;
 
-  if(fscanf(f, "TRF%hhu\n", &version)!=LIBVIDSTAB_FILE_FORMAT_VERSION) goto parse_error_handling;
+  if(fscanf(f, "TRF%hhu", &version)!=LIBVIDSTAB_FILE_FORMAT_VERSION) goto parse_error_handling;
   if(readInt32(&conf.accuracy, f)<=0) goto parse_error_handling;
   if(readInt32(&conf.shakiness, f)<=0) goto parse_error_handling;
   if(readInt32(&conf.stepSize, f)<=0) goto parse_error_handling;
diff --git a/src/transform.c b/src/transform.c
index 380ad28..22f5932 100644
--- a/src/transform.c
+++ b/src/transform.c
@@ -32,7 +32,9 @@
 #endif
 
 #include <math.h>
+#ifndef _MSC_VER
 #include <libgen.h>
+#endif
 #include <string.h>
 
 const char* interpol_type_names[5] = {"No (0)", "Linear (1)", "Bi-Linear (2)",
@@ -139,7 +141,7 @@ int vsTransformPrepare(VSTransformData* td, const VSFrame* src, VSFrame* dest){
   // with the transformed version
   td->dest = *dest;
   if(src==dest || td->srcMalloced){ // in place operation: we have to copy the src first
-    if(vsFrameIsNull(&td->src)) {
+    if(!td->srcMalloced) {
       vsFrameAllocate(&td->src,&td->fiSrc);
       td->srcMalloced = 1;
     }
diff --git a/src/transform.h b/src/transform.h
index 62e8ff4..2e8f086 100644
--- a/src/transform.h
+++ b/src/transform.h
@@ -25,7 +25,9 @@
 #define __TRANSFORM_H
 
 #include <math.h>
+#ifndef _MSC_VER
 #include <libgen.h>
+#endif
 #include <stdint.h>
 #include "transformtype.h"
 #include "frameinfo.h"
diff --git a/src/transformfixedpoint.c b/src/transformfixedpoint.c
index 3549747..b36a620 100644
--- a/src/transformfixedpoint.c
+++ b/src/transformfixedpoint.c
@@ -56,7 +56,7 @@
 
 /** interpolateBiLinBorder: bi-linear interpolation function that also works at the border.
     This is used by many other interpolation methods at and outsize the border, see interpolate */
-inline void interpolateBiLinBorder(uint8_t *rv, fp16 x, fp16 y,
+void interpolateBiLinBorder(uint8_t *rv, fp16 x, fp16 y,
                                    const uint8_t *img, int img_linesize,
                                    int32_t width, int32_t height, uint8_t def)
 {
@@ -119,7 +119,7 @@ inline static short bicub_kernel(fp16 t, short a0, short a1, short a2, short a3)
 }
 
 /** interpolateBiCub: bi-cubic interpolation function using 4x4 pixel, see interpolate */
-inline void interpolateBiCub(uint8_t *rv, fp16 x, fp16 y,
+void interpolateBiCub(uint8_t *rv, fp16 x, fp16 y,
                              const uint8_t *img, int img_linesize,
                              int width, int height, uint8_t def)
 {
@@ -159,7 +159,7 @@ inline void interpolateBiCub(uint8_t *rv, fp16 x, fp16 y,
 
 
 /** interpolateBiLin: bi-linear interpolation function, see interpolate */
-inline void interpolateBiLin(uint8_t *rv, fp16 x, fp16 y,
+void interpolateBiLin(uint8_t *rv, fp16 x, fp16 y,
                              const uint8_t *img, int img_linesize,
                              int32_t width, int32_t height, uint8_t def)
 {
@@ -187,7 +187,7 @@ inline void interpolateBiLin(uint8_t *rv, fp16 x, fp16 y,
 }
 
 /** interpolateLin: linear (only x) interpolation function, see interpolate */
-inline void interpolateLin(uint8_t *rv, fp16 x, fp16 y,
+void interpolateLin(uint8_t *rv, fp16 x, fp16 y,
                            const uint8_t *img, int img_linesize,
                            int width, int height, uint8_t def)
 {
@@ -205,7 +205,7 @@ inline void interpolateLin(uint8_t *rv, fp16 x, fp16 y,
 }
 
 /** interpolateZero: nearest neighbor interpolation function, see interpolate */
-inline void interpolateZero(uint8_t *rv, fp16 x, fp16 y,
+void interpolateZero(uint8_t *rv, fp16 x, fp16 y,
                             const uint8_t *img, int img_linesize,
                             int width, int height, uint8_t def)
 {
@@ -230,7 +230,7 @@ inline void interpolateZero(uint8_t *rv, fp16 x, fp16 y,
  *            def: default value if coordinates are out of range
  * Return value:  None
  */
-inline void interpolateN(uint8_t *rv, fp16 x, fp16 y,
+void interpolateN(uint8_t *rv, fp16 x, fp16 y,
                          const uint8_t *img, int img_linesize,
                          int width, int height,
                          uint8_t N, uint8_t channel,
@@ -373,6 +373,9 @@ int transformPlanar(VSTransformData* td, VSTransform t)
      *  t the translation, and M the rotation and scaling matrix
      *      p_s = M^{-1}(p_d - c_d - t) + c_s
      */
+#ifdef USE_OMP
+#pragma omp parallel for private(x) schedule(static)
+#endif
     for (y = 0; y < dh; y++) {
       // swapping of the loops brought 15% performace gain
       int32_t y_d1 = (y - c_d_y);
diff --git a/src/vidstabdefines.h b/src/vidstabdefines.h
index 4e8f23a..c4ce679 100644
--- a/src/vidstabdefines.h
+++ b/src/vidstabdefines.h
@@ -85,13 +85,9 @@ extern int VS_ERROR;
 extern int VS_OK;
 
 
-#define vs_log_error(tag, format, args...) \
-    vs_log(VS_ERROR_TYPE, tag, format , ## args)
-#define vs_log_warn(tag, format, args...) \
-    vs_log(VS_WARN_TYPE, tag, format , ## args)
-#define vs_log_info(tag, format, args...) \
-    vs_log(VS_INFO_TYPE, tag, format , ## args)
-#define vs_log_msg(tag, format, args...) \
-    vs_log(VS_MSG_TYPE, tag, format , ## args)
+#define vs_log_error(...) vs_log(VS_ERROR_TYPE, __VA_ARGS__)
+#define vs_log_warn(...) vs_log(VS_WARN_TYPE, __VA_ARGS__)
+#define vs_log_info(...) vs_log(VS_INFO_TYPE, __VA_ARGS__)
+#define vs_log_msg(...) vs_log(VS_MSG_TYPE, __VA_ARGS__)
 
 #endif /* VIDSTABDEFINES_H_ */
